job-server/src/main/scala/spark/jobserver/io/FileCacher.scala

```diff
@@ -27,21 +27,45 @@ trait FileCacher {
     appName + "-" + uploadTime.toString("yyyyMMdd_hhmmss_SSS") + s".${binaryType.extension}"
   }
 
+  // Random temporary filename suffix possible chars
```

Comment:
I wanted to have more control over the file created (also having future development in mind), but as long as I can control the target directory ( https://docs.oracle.com/javase/8/docs/api/java/io/File.html#createTempFile-java.lang.String-java.lang.String-java.io.File- ) I can and will change it to the Java implementation.